src/cluster_undersample/cluster_undersample.jl

```diff
@@ -0,0 +1,207 @@
+"""
+Assuming that all the observations in the observation matrix X belong to the same class, undersample
+the observations using clustering undersampling
+
+# Arguments
+- `X`: A matrix where each column is an observation of floats
+- `n`: Number of observations in the undersampled dataset
+- `mode::AbstractString="nearest`: If `center` then the undersampled data will consist of the centriods of 
+    each cluster found. Meanwhile, if `nearest` then it will consist of the nearest neighbor of each centroid.
+- `maxiter::Integer=100`: Maximum number of iterations to run K-means
+- `rng::Integer=42`: Random number generator seed
```

Comment:
It seems that previously we allowed an arbitrary `AbstractRNG` in addition to an integer seed for `Xoshiro`, no? It is more Julian to allow any RNG, and allowing an integer is really just a convenience feature.

Reply:
Indeed, I just earlier thought this wasn't possible. Fixed it now.